src/clients/python/main.py

```diff
@@ -0,0 +1,389 @@
+        url += "-linux"
+    elif platform.system() == "Windows":
+        url += "-windows.exe"
+    else:
+        raise ValueError("Unsupported OS")
+
+    url += flags
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, follow_redirects=True)
+        response.raise_for_status()
+
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    async with aiofiles.open(file_path, "wb") as f:
+        await f.write(response.content)
+
+    os.chmod(file_path, 0o755)
+    return str(file_path)
+
+
+def get_cache_dir() -> Path:
+    if platform.system() == "Darwin":
+        return Path.home() / "Library" / "Caches" / "python-webview"
+    elif platform.system() == "Linux":
+        return Path.home() / ".cache" / "python-webview"
+    elif platform.system() == "Windows":
+        return Path(os.environ["LOCALAPPDATA"]) / "python-webview" / "Cache"
+    else:
+        raise ValueError("Unsupported OS")
+
+
+class WebView:
+    process: subprocess.Popen[bytes] | None = None
+
+    def __init__(self, options: WebViewOptions, webview_binary_path: str | None = None):
+        self.options = options
+        if webview_binary_path is not None:
+            self.__start_process(webview_binary_path)
+        self.internal_event = AsyncIOEventEmitter()
+        self.external_event = AsyncIOEventEmitter()
+        self.buffer = b""
+
+    def __start_process(self, webview_binary_path: str):
+        encoded_options = str(msgspec.json.encode(self.options), "utf-8")
+        self.process = subprocess.Popen(
+            [webview_binary_path, encoded_options],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,  # Capture stderr properly
+            text=False,
+            bufsize=0,
+            env=os.environ,
+        )
+        assert self.process.stdin is not None
+        assert self.process.stdout is not None
+        assert self.process.stderr is not None
+
+        # Create StreamReader for non-blocking reads
+        loop = asyncio.get_event_loop()
+        self.stdout_reader = asyncio.StreamReader()
+        protocol = asyncio.StreamReaderProtocol(self.stdout_reader)
+        loop.create_task(loop.connect_read_pipe(lambda: protocol, self.process.stdout))
+
+        # Also handle stderr
+        self.stderr_reader = asyncio.StreamReader()
+        stderr_protocol = asyncio.StreamReaderProtocol(self.stderr_reader)
+        loop.create_task(
+            loop.connect_read_pipe(lambda: stderr_protocol, self.process.stderr)
+        )
+        loop.create_task(self._pipe_stderr())
+
+    async def _pipe_stderr(self):
+        """Pipe stderr from the subprocess to Python's stderr"""
+        while True:
+            try:
+                line = await self.stderr_reader.readline()
+                if not line:
+                    break
+                sys.stderr.buffer.write(line)
+                sys.stderr.buffer.flush()
+            except Exception:
```

Comment:
You almost never want to be doing a generic `except Exception`. Instead, write this for the exact Exception you'd like to catch. This means:

- As a reader, I know what to expect when I want to use this function
- Any unhandled exception is raised instead of the `break` and being ignored silently